cogs/moveusers.py

```diff
@@ -1,13 +1,95 @@
-        await member.move_to(channel)
-    
+    async def move(self, ctx, member: discord.Member, channel: str):
+        destination_channel = self.find_similar_channel(channel, ctx.guild.voice_channels)
+        await member.move_to(destination_channel)
+
+
+    @commands.command()
+    async def automove(self, ctx):
+        voice_channels = ctx.guild.voice_channels
+        # Group users by their activities in voice channels
+        voice_channel_groups = {}
+        for channel in voice_channels:
+            activity_count = {"Streaming": 0, "Nothing": 0}
+            for member in channel.members:
+                if member.activity and member.activity.name != "Spotify":
+                    activity_name = member.activity.name
+                else:
+                    activity_name = "Nothing"
+
+                if member.voice and member.voice.self_stream:
+                    activity_count["Streaming"] += 1
+
+                if activity_name in activity_count:  # Check if activity exists in the dictionary
+                    activity_count[activity_name] += 1
+                else:
+                    activity_count[activity_name] = 1
+
+            voice_channel_groups[channel.name] = activity_count
+
+        print(voice_channel_groups)
+
+        # Movement
+        for channel in voice_channel_groups:  # Loop through voice channel groups
+            current_channel = discord.utils.get(ctx.guild.voice_channels, name=channel)
+
+            if voice_channel_groups[channel]["Streaming"] > 0: # if there is at least one person streaming
+                members_to_move = current_channel.members
+                if "Visual Studio Code" in voice_channel_groups[channel]:
+                    if self.bot.get_channel(DIFFERENT_NAME_CHANNELS["Study"]).voice_states:
+                        await self.move_users_to_channel_by_id(members_to_move, DIFFERENT_NAME_CHANNELS["Lounge"])
+                    else:
+                        await self.move_users_to_channel_by_id(members_to_move, DIFFERENT_NAME_CHANNELS["Study"])
+                else:
+                    if self.bot.get_channel(DIFFERENT_NAME_CHANNELS["Cinema"]).voice_states:
+                        await self.move_users_to_channel_by_id(members_to_move, DIFFERENT_NAME_CHANNELS["Lounge"])
+                    else:
+                        await self.move_users_to_channel_by_id(members_to_move, DIFFERENT_NAME_CHANNELS["Cinema"])
+            
+            elif voice_channel_groups[channel]["Nothing"] == len(current_channel.members): # If noone is playing a game
+                members_to_move = current_channel.members
+                await self.move_users_to_channel_by_id(members_to_move, DIFFERENT_NAME_CHANNELS["Lounge"])
+            
+            elif len(voice_channel_groups[channel]) == 3 and voice_channel_groups[channel][2] == 1: # If one person is playing a game and everyone else is not
```

Comment:
Hard coded numbers are not a good idea here.

Reply:
Completely fine considering the structure of the dictionary here.

Reply:
Not necessarily. Dictionaries are not sorted and have no sense of ordering so we cannot guarantee that `voice_channel_groups[channel][2]` corresponds to some game and not `Nothing` or `Streaming`.

Reply:
So the keys and their corresponding values would be shifted around?

Reply:
Typically no but python makes no guarantees for dictionaries so yes there is a possibility that it can be shifted around.